cpu/cc2538/dev/uart.c

```diff
@@ -136,7 +136,9 @@
  * Baud rate defines used in uart_init() to set the values of UART_IBRD and
  * UART_FBRD in order to achieve the configured baud rates.
  */
-#define UART_CLOCK_RATE       16000000 /* 16 MHz */
+
+#define UART_CLOCK_RATE			(SYS_CTRL_32MHZ/(1 << SYS_CTRL_SYS_DIV))
+
```

Comment:
Yes, it seems to come from the 5/3 rule tested in `sys-ctrl.h` for an I/O clock of 16 MHz. And 16 MHz for the SPI currently configured to use the I/O clock (page 433). Is that a problem?

The reason for choosing the I/O clock is explained in table 7-2 and in §7.3.1 (page 195). With the current `enter_pm0()`, choosing the I/O clock does not matter much, and it even prevents users from selecting slow clocks in run mode as you pointed out. However, in the future, that would allow to slow the system clock down during PM0, without disturbing UART and SSI operation, thus improving power consumption without altering the behavior.

So let's stick to the system clock for UART and SPI for now, and we'll switch to the I/O clock if `enter_pm0()` is improved in the future:
```
#define UART_CLOCK_RATE       SYS_CTRL_SYS_CLOCK
```

So the SPI driver also has to be updated to use the system clock. The 5/3 rule tested in `sys-ctrl.h` is UART-specific, so it should be dropped.